kubermatic/kubermatic
# Deploy gatekeeper

**What this PR does / why we need it**:
This PR adds gatekeeper deployment as part of the Kubermatic OPA integration

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #5784 

**Documentation:**
https://github.com/kubermatic/kubermatic/pull/5768/files


**Special notes for your reviewer**:
Tested on dev cluster. Testing procedure is like this:
1. Run local `seed-controller` and `api` with the same `worker-name` arg set.
2. Run local `dashboard`.
3. Create a new cluster.
4. When the cluster is ready (you can already check if the seed resources are deployed correctly), edit is `user-cluster-controller` deployment and set `replicas` to 0.
5. Run your local `user-cluster-controller` with the arguments set for the the proper user cluster - hint , you can check what args the original `user-cluster-controller` has.
6. Check to see if the gatekeeper resources are deployed on user cluster.
7. You can use some of the constraint templates from the gatekeeper demo to check if its working correctly - https://github.com/open-policy-agent/gatekeeper/tree/master/demo/basic. Keep in mind to create ConstraintTemplates and Constraints and Configs in the user cluster.

An addtional step would be to remove the addon gatekeeper installation option. Let me know if you want it in this PR or it can be a separate task.

**Does this PR introduce a user-facing change?**:

```release-note
Gatekeeper is now deployed automatically for the user clusters as part of Kubermatic OPA integration
```
